File: net/management-api-v2/PatchAssetFolders.cs
// DocSection: cm_api_v2_patch_asset_folders
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kentico.Kontent.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    ProjectId = "<YOUR_PROJECT_ID>"
});

var response = await client.ModifyAssetFoldersAsync(new AssetFolderOperationBaseModel[]
{
    new AssetFolderAddIntoModel
    {
        Reference = Reference.ByExternalId("folder-with-shared-asset"),
        Value = new AssetFolderHierarchy
        {
            ExternalId = "folder-with-shared-assets",
            Name = "Shared assets",
            Folders = Enumerable.Empty<AssetFolderHierarchy>(),
        },
        Before = Reference.ByExternalId("folder-with-downloadable-assets")
    },
    new AssetFolderRemoveModel
    {
        Reference = Reference.ByExternalId("folder-with-archived-assets")
    },
    new AssetFolderRenameModel
    {
        Reference = Reference.ByExternalId("folder-documents"),
        Value = "Legal documents"
    }
});
// EndDocSection

File: js/management-api-v2/PatchAssetFolders.js
// DocSection: cm_api_v2_patch_asset_folders
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
// Using ES6 syntax
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  environmentId: '<YOUR_ENVIRONMENT_ID>,
  apiKey: '<YOUR_API_KEY>'
});

const response = await client.modifyAssetFolders()
  .withData(
    [
      {
        op: "addinto",
        value: {
          external_id: "folder-with-shared-assets",
          name: "Shared assets",
          folders: []
        },
        before: {
          external_id: "folder-with-downloadable-assets"
        }
      },
      {
        op: "remove",
        reference: {
          external_id: "folder-with-downloadable-archives"
        }
      },
      {
        op: "rename",
        value: "Legal documents",
        reference: {
          external_id: "folder-documents"
        }
      }
    ]
  )
  .toPromise();
// EndDocSection
